nibabel/streamlines/trk.py

```diff
@@ -0,0 +1,520 @@
+            self.offset_data = f.tell()
+
+    def __iter__(self):
+        i4_dtype = np.dtype(self.endianness + "i4")
+        f4_dtype = np.dtype(self.endianness + "f4")
+
+        with Opener(self.fileobj) as f:
+            start_position = f.tell()
+
+            nb_pts_and_scalars = int(3 + self.header[Field.NB_SCALARS_PER_POINT])
+            pts_and_scalars_size = int(nb_pts_and_scalars * f4_dtype.itemsize)
+
+            slice_pts_and_scalars = lambda data: (data, [])
+            if self.header[Field.NB_SCALARS_PER_POINT] > 0:
+                # This is faster than `np.split`
+                slice_pts_and_scalars = lambda data: (data[:, :3], data[:, 3:])
+
+            # Using np.fromfile would be faster, but does not support StringIO
+            read_pts_and_scalars = lambda nb_pts: slice_pts_and_scalars(np.ndarray(shape=(nb_pts, nb_pts_and_scalars),
+                                                                                   dtype=f4_dtype,
+                                                                                   buffer=f.read(nb_pts * pts_and_scalars_size)))
+
+            properties_size = int(self.header[Field.NB_PROPERTIES_PER_STREAMLINE] * f4_dtype.itemsize)
+            read_properties = lambda: []
+            if self.header[Field.NB_PROPERTIES_PER_STREAMLINE] > 0:
+                read_properties = lambda: np.fromstring(f.read(properties_size),
+                                                        dtype=f4_dtype,
+                                                        count=self.header[Field.NB_PROPERTIES_PER_STREAMLINE])
+
+            # Set the file position at the beginning of the data.
+            f.seek(self.offset_data, os.SEEK_SET)
+
+            # If 'count' field is 0, i.e. not provided, we have to loop until the EOF.
+            nb_streamlines = self.header[Field.NB_STREAMLINES]
+            if nb_streamlines == 0:
+                nb_streamlines = np.inf
+
+            i = 0
+            while i < nb_streamlines:
+                nb_pts_str = f.read(i4_dtype.itemsize)
+
+                # Check if we reached EOF
+                if len(nb_pts_str) == 0:
+                    break
+
+                # Read number of points of the next streamline.
+                nb_pts = struct.unpack(i4_dtype.str[:-1], nb_pts_str)[0]
+
+                # Read streamline's data
+                pts, scalars = read_pts_and_scalars(nb_pts)
+                properties = read_properties()
+
+                # TRK's streamlines are in 'voxelmm' space, we send them to voxel space.
+                pts = pts / self.header[Field.VOXEL_SIZES]
+                # TrackVis considers coordinate (0,0,0) to be the corner of the
+                # voxel whereas streamlines returned assume (0,0,0) to be the
+                # center of the voxel. Thus, streamlines are shifted of half
+                #a voxel.
+                pts -= np.array(self.header[Field.VOXEL_SIZES])/2.
```

Comment:
Is this right? after the points have been divided by `VOXEL_SIZE` don't you just want to do `-.5`?